### PR TITLE
implement Dbus updateAccount and listDevices

### PR DESCRIPTION
**Pull request by johnfreed**
implement updateAccount(deviceName\<s\>) to change device name

implement listDevices

update documentation

### CODE REVIEW

File: src/main/java/org/asamk/Signal.java [hunk 1]
```diff
@@ -83,6 +83,14 @@ long sendGroupMessageReaction(
 
     boolean isRegistered();
 
+    void addDevice(String uri) throws Error.Failure;
+
+    void removeDevice(int deviceId) throws Error.Failure;
+
+    List<String> listDevices() throws Error.Failure;
+
+    void updateAccount(String deviceName) throws Error.Failure;
```
Comment (AsamK) [inline src/main/java/org/asamk/Signal.java:92]: I think for the dbus interface a better name would be `updateDeviceName`

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java [hunk 2]
```diff
@@ -60,6 +64,54 @@ public String getObjectPath() {
...
     }
 
+    @Override
+    public void addDevice(String uri) {
+        try {
+            m.addDeviceLink(new URI(uri));
+        } catch (IOException | InvalidKeyException e) {
+            throw new Error.Failure(e.getClass().getSimpleName() + " Add device link failed. " + e.getMessage());
+        } catch (URISyntaxException e) {
+            throw new Error.Failure(e.getClass().getSimpleName() + " Device link uri has invalid format: " + e.getMessage());
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java:74]: Maybe add a new error class for this? `Error.InvalidUri`

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java [hunk 3]
```diff
@@ -60,6 +64,54 @@ public String getObjectPath() {
...
+        } catch (IOException | Error.Failure e) {
+            throw new Error.Failure("Failed to get linked devices: " + e.getMessage());
+        }
+
+        for (var d : devices) {
+            var name = d.getName();
+            if (name == null) {name = "null";}
+            results.add(name);
+        }
+        return results;
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java:103]: You could replace this for loop with a stream.
```suggestion
        return devices.stream()
            .map(d -> d.getName() == null ? "" : d.getName())
            .collect(Collectors.toList());
```

File: man/signal-cli-dbus.5.adoc [hunk 4]
```diff
@@ -29,126 +29,252 @@ method(arg1<type>, arg2<type>, ...) -> return<type>
...
+* <x>   : Signed 64-bit integer (long)
+* <i>   : Signed 32-bit integer (int)
 * <>    : no return value
 
 Exceptions are the names of the Java Exceptions returned in the body field. They typically contain an additional message with details. All Exceptions begin with "org.asamk.Signal.Error." which is omitted here for better readability.
 
 Phone numbers always have the format +<countrycode><regional number>
 
-== Methods
+=== Control methods
```
Comment (AsamK) [inline man/signal-cli-dbus.5.adoc:44]: Thanks for documenting the Control interface!

File: man/signal-cli-dbus.5.adoc [hunk 5]
```diff
@@ -29,126 +29,252 @@ method(arg1<type>, arg2<type>, ...) -> return<type>
...
+Exceptions: None
 
-sendMessage(message<s>, attachments<as>, recipient<s>) -> timestamp<x>::
-sendMessage(message<s>, attachments<as>, recipients<as>) -> timestamp<x>::
-* message     : Text to send (can be UTF8)
-* attachments : String array of filenames to send as attachments (passed as filename, so need to be readable by the user signal-cli is running under)
-* recipient   : Phone number of a single recipient
-* recipients  : Array of phone numbers 
-* timestamp   : Can be used to identify the corresponding signal reply
+getGroupIdsBase64() -> groupList<as>::
```
Comment (AsamK) [inline man/signal-cli-dbus.5.adoc:115]: Please remove the docs for methods from the other PRs here.

**Review by AsamK — CHANGES_REQUESTED** [review]

File: man/signal-cli-dbus.5.adoc (hunk 5 above)
Comment (johnfreed) [reply man/signal-cli-dbus.5.adoc]: done

File: man/signal-cli-dbus.5.adoc (hunk 4 above)
Comment (johnfreed) [reply man/signal-cli-dbus.5.adoc:47]: you're welcome

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java (hunk 3 above)
Comment (johnfreed) [reply src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java]: syntax still baffles me, but hey it works :)

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java (hunk 2 above)
Comment (johnfreed) [reply src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java]: and activated :)

**Comment by AsamK** [comment]
I've refactored the dbus devices interface to export a separate dbus object for each device. Should make it easier in the future to extend the interface.

**Pull request merged**